google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/internal/RowSetUtil.java

```diff
@@ -125,6 +166,67 @@ private static RowRange truncateRange(RowRange range, ByteString split, boolean
     return newRange.build();
   }
 
+  public static List<RowRange> splitRange(RowRange range, ByteString split, boolean fromStart) {
+    List<RowRange> rowRangesList = new ArrayList<RowRange>();
+
+    if (fromStart) {
+      // range end is on or left of the split: skip
+      if (EndPoint.extract(range).compareTo(new EndPoint(split, true)) < 0) {
+        rowRangesList.add(range);
+        return rowRangesList;
+      }
+    } else {
+      // range is on or right of the split
+      if (StartPoint.extract(range).compareTo(new StartPoint(split, true)) > 0) {
+        rowRangesList.add(range);
+        return rowRangesList;
+      }
+      // either set = or set the range's start & end key acc to you
+    }
+
+    if (fromStart) {
+      // range start is on or left of the split
+      if (StartPoint.extract(range).compareTo(new StartPoint(split, true)) < 0) {
+        RowRange beforeSplitKeyRange =
+            range
+                .toBuilder()
+                .setStartKeyOpen(
+                    range.getStartKeyOpen().isEmpty()
+                        ? range.getStartKeyClosed()
+                        : range.getStartKeyOpen())
+                .setEndKeyOpen(split)
+                .build();
+        rowRangesList.add(beforeSplitKeyRange);
+        // RowRange afterSplitKeyRange = range.toBuilder().setStartKeyOpen(split).build();
+      }
+      if (EndPoint.extract(range).compareTo(new EndPoint(split, true)) > 0) {
```

Comment:
this statement is already covered by the previous block (line 174 to 176). so it's redundant here. we should always add RowRange afterSplitKeyRange = range.toBuilder().setStartKeyOpen(split);